p2p/host/eventbus/basic.go

```diff
@@ -360,7 +374,11 @@ func (n *wildcardNode) emit(evt interface{}) {
 		// record channel full events before blocking
 		sendSubscriberMetrics(n.metricsTracer, sink)
 
-		sink.ch <- evt
+		select {
+		case sink.ch <- evt:
+		default:
+			n.slowConsumerTimer = emitAndLogError(n.slowConsumerTimer, wildcardType, evt, sink)
+		}
 	}
 	n.RUnlock()
```

Comment:
This won't work with an `RLock`, right? We need Lock since we're updating state on `n`

I think it's fine to create the timer right when we create the node. I don't think there are that many nodes in total.

Reply:
I didn't realize this was an `RLock`. Thanks for highlighting that. I've updated the code to set the timer within a lock. It's technically possible to allocate two or more stopped timers, but that should be fine as it'll get cleaned up by the GC.
 
> I think it's fine to create the timer right when we create the node.

I considered this, but there's no way to create a stopped timer. You must create an armed one. So if you are going through the trouble of setting one, you may as well use it.